### ISSUE TITLE
The KatharsisRegistryConfiguration class do not use the serialization configuration of Spring Boot

### ISSUE THREAD

**Issue by olabomba** [katharsis-project/katharsis-spring#25]
If I want to use a custom serialization option (for example, by using : spring.jackson.property-naming-strategy), Katharsis do not use it.

I think we could inject the ObjectMapper in KatharsisRegistryConfiguration (with Autowired), and use the config with : new ResourceFieldNameTransformer(objectMapper.getSerializationConfig()))

I have not gone further because I need a serialization option that comes with Jackson 2.7, and Katharsis-core is not compatible with this version (I'll go up another issue).

Thanks.

**Comment by meshuga** [comment]
Since katharsis-project/katharsis-core#280 has been solved, could you make a snippet of code of how to autowire `ObjectMapper` and pass serialization config to Katharsis? I've tried it, but then Spring had problems with proper `JacksonConfiguration` registration.

**Comment by olabomba** [comment]
Ok, thank you so much :)
I just tried with your correction, and a simple configuration like this:

    @Autowired 
    private ObjectMapper objectMapper 

Then :

    @Bean 
    public ResourceRegistry resourceRegistry() { 
        ResourceRegistryBuilder registryBuilder = 
            new ResourceRegistryBuilder(serviceLocator, 
                new ResourceInformationBuilder( 
                    new ResourceFieldNameTransformer(objectMapper.getSerializationConfig()))) 
 
        String serverUri = properties.getDomainName() + properties.getPathPrefix(); 
        ResourceRegistry resourceRegistry = registryBuilder.build(properties.getResourcePackage(), serverUri); 
        return resourceRegistry 
    } 

This is working well with my serialization option (KEBAB CASE), what does not work in your favor?

**Comment by masterspambot** [comment]
:+1:

**Comment by meshuga** [comment]
I'm getting circular dependency exception because `ResourceRegistry` requires an instance of `ObjectMapper` that requires `Module` from `parameterNamesModule` which requires `ResourceRegistry` and I don't know what to do :disappointed: 

```java
@Configuration
public class JacksonConfiguration {

    @Autowired
    private ResourceRegistry resourceRegistry;

    @Bean
    public Module parameterNamesModule() {
        JsonApiModuleBuilder jsonApiModuleBuilder = new JsonApiModuleBuilder();
        return jsonApiModuleBuilder.build(resourceRegistry);
    }
}
```

and 

```java
@Configuration
@EnableConfigurationProperties(KatharsisSpringBootProperties.class)
public class KatharsisRegistryConfiguration {

    @Autowired
    private KatharsisSpringBootProperties properties;

    @Autowired
    private SpringServiceLocator serviceLocator;

    @Autowired
    private ObjectMapper objectMapper;

    @Bean
    public ResourceRegistry resourceRegistry() {
        ResourceRegistryBuilder registryBuilder =
            new ResourceRegistryBuilder(serviceLocator,
                new ResourceInformationBuilder(
                    new ResourceFieldNameTransformer(objectMapper.getSerializationConfig())));

        String serverUri = properties.getDomainName() + properties.getPathPrefix();
        ResourceRegistry resourceRegistry = registryBuilder.build(properties.getResourcePackage(), serverUri);
        return resourceRegistry;
    }
}
```

**Comment by masterspambot** [comment]
![](https://media0.giphy.com/media/EjzJxwcqpO2ly/200_s.gif)

**Comment by olabomba** [comment]
I just tested the 2.3.2 and everything works perfectly; So I guess the circular dependency problem is solved? :)
(And again, thank you)

**Comment by andy36** [comment]
I was attempting the same thing and wanted to share my (successful) results.
I was using the spring boot jackson attributes to set the naming strategy and created my own KatharsisRegistryConfiguration as above to use the ObjectMapper.getSerializationConfig
When doing this I noticed that my outputs stopped being json api and were just json. The parameterNamesModule module wasn't being loaded or used anymore. 
I fixed this by just overriding the whole Filter and manually adding the module as below.

@Bean
 public Filter springBootSampleKatharsisFilter() {
      objectMapper.registerModule(parameterNamesModule);
      return new KatharsisFilterV2(objectMapper, queryParamsBuilder, resourceRegistry, requestDispatcher,
            properties.getPathPrefix());
  }
